### PR TITLE
[1/N] replace THPUtils_assert with TORCH_CHECK

### PR DESCRIPTION
**Pull request by cyyever**
This PR replaces THPUtils_assert with TORCH_CHECK.

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
tbh we should just have deleted this API and used TORCH_CHECK (or the appropriate variant to for custom types of errors) throughout.
That would actually simplify all of this instead of adding a new API.
WDYT?

**Comment by cyyever** [comment]
@albanD I would like to unify all these macros, but I would add a TORCH_CHECK_FMT for libfmt to avoid breaking code.

**Comment by albanD** [comment]
IIRC TORCH_CHECK is plain better and we just never had a chance to finish removing all the other asserts.

**Comment by cyyever** [comment]
I see. There are multiple assertion macros in PyTorch. Sometimes I don't know which to use.

File: torch/csrc/utils.h [hunk 1]
```diff
@@ -155,11 +155,16 @@
 #define THPUtils_assert(cond, ...) \
   THPUtils_assertRet(nullptr, cond, __VA_ARGS__)
 #define THPUtils_assertRet(value, cond, ...) \
-  if (THP_EXPECT(!(cond), 0)) {              \
-    THPUtils_setError(__VA_ARGS__);          \
+  if (C10_UNLIKELY_OR_CONST(!(cond))) {                               \
+    try { \
+      TORCH_CHECK((cond), ##__VA_ARGS__);  \
+    } catch (const std::exception& e) { \
+      THPUtils_setError(e.what());          \
```
Comment (albanD) [inline torch/csrc/utils.h:162]: Most of the use above are already guarded with HANDLE_TH_ERRORS which will do this properly. No need to set the python error here.
For the few other cases, I think we can just add the HANDLE_TH_ERRORS macro where it is missing.

Comment (cyyever) [reply torch/csrc/utils.h:162]: OK, I see.

**Review by albanD — APPROVED** [review]
Awesome!
Thanks for the update!

**Pull request closed without merging**